addOns/commonlib/src/test/java/org/zaproxy/addon/commonlib/ResourceidentificationUtilsUnitTest.java

```diff
@@ -0,0 +1,164 @@
+            strings = {
+                "any.css",
+                "any.js",
+                "any.html",
+                "any.txt",
+                "any.xml",
+                "any.json",
+                "any.jpeg",
+                "any.png",
+                "any.gif"
+            })
+    void shouldReturnFalseWhenRequestUrlDoesNotSeemToBeAFontFile(String fileName)
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        HttpMessage msg = new HttpMessage();
+        msg.getRequestHeader().setURI(new URI("http://example.com/" + fileName, false));
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isFont(msg), is(equalTo(false)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"something.js", "something.jsx"})
+    void shouldReturnTrueWhenRequestUrlSeemsToBeAJavaScriptFile(String fileName)
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.getRequestHeader().setURI(new URI("http://example.com/" + fileName, false));
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isJavaScript(msg), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldReturnTrueWhenResponseSeemsToBeAJavaScriptFile()
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.setResponseBody("<script>alert('hello');</script>");
+        msg.getRequestHeader().setURI(new URI("http://example.com/any.js", true));
+        msg.getResponseHeader().setHeader(HttpResponseHeader.CONTENT_TYPE, "text/javascript");
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isJavaScript(msg), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldReturnTrueWhenRequestUrlSeemsToBeACssFile()
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.getRequestHeader().setURI(new URI("http://example.com/something.css", false));
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isCss(msg), is(equalTo(true)));
+    }
+
+    @Test
+    void shouldReturnTrueWhenResponseSeemsToBeACssFile()
+            throws HttpMalformedHeaderException, URIException, NullPointerException {
+        // Given
+        msg.setResponseBody(
+                "body {background-color: powderblue;}\n"
+                        + "h1 {color: blue;}\n"
+                        + "p {color: red;"
+                        + "margin-left:85.36370249136206%"
+                        + "}");
+        msg.getRequestHeader().setURI(new URI("http://example.com/any.css", true));
+        msg.getResponseHeader().setHeader(HttpResponseHeader.CONTENT_TYPE, "text/css");
+        // When / Then
+        assertThat(ResourceIdentificationUtils.isCss(msg), is(equalTo(true)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"hello_world.pdf", "hello_world.png", "hello_world.xlsx"})
```

Comment:
Could just hardcode a string with control chars.

Reply:
I just figured this is more representative of what’s actually encountered.

Reply:
It might be but it's not necessary to unit test the class.

Reply:
I'll tiddy this up shortly.